res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,974 @@
+        engine.setValue(field.group, "reloop_andstop", field.value);
+    } else {
+        engine.setValue(field.group, "reloop_toggle", field.value);
+    }
+};
+
+TraktorS2MK3.beatjumpHandler = function (field) {
+    var delta = 1;
+    if ((field.value + 1) % 16 === TraktorS2MK3.beatjumpState[field.group]) {
+        delta = -1;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        var beatjump_size = engine.getValue(field.group, "beatjump_size");
+        if (delta > 0) {
+            engine.setValue(field.group, "beatjump_size", beatjump_size * 2);
+        } else {
+            engine.setValue(field.group, "beatjump_size", beatjump_size / 2);
+        }
+    } else {
+        if (delta < 0) {
+            engine.setValue(field.group, "beatjump_backward", 1);
+            engine.setValue(field.group, "beatjump_backward", 0);
+        } else {
+            engine.setValue(field.group, "beatjump_forward", 1);
+            engine.setValue(field.group, "beatjump_forward", 0);
+        }
+    }
+
+    TraktorS2MK3.beatjumpState[field.group] = field.value;
+};
+
+TraktorS2MK3.quantizeHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    var res = !(engine.getValue("[Channel1]", "quantize") && engine.getValue("[Channel2]", "quantize"));
+    engine.setValue("[Channel1]", "quantize", res);
+    engine.setValue("[Channel2]", "quantize", res);
+    TraktorS2MK3.outputHandler(res, field.group, "quantize");
+};
+
+TraktorS2MK3.microphoneHandler = function (field) {
+    if (field.value) {
+        if (TraktorS2MK3.microphonePressedTimer === 0) {
+            // Start timer to measure how long button is pressed
+            TraktorS2MK3.microphonePressedTimer = engine.beginTimer(1000, "TraktorS2MK3.microphoneTimer()");
+        }
+
+        var talkover = engine.getValue("[Microphone]", "talkover");
+        engine.setValue("[Microphone]", "talkover", !talkover);
+    } else {
+        // Button is released, check if timer is still running
+        if (TraktorS2MK3.microphonePressedTimer !== 0) {
+            // short klick -> permanent activation
+            engine.stopTimer(TraktorS2MK3.microphonePressedTimer);
+            TraktorS2MK3.microphonePressedTimer = 0;
+        } else {
+            engine.setValue("[Microphone]", "talkover", 0);
+        }
+    }
+};
+
+TraktorS2MK3.microphoneTimer = function () {
+    // Reset microphone button timer if active
+    if (TraktorS2MK3.microphonePressedTimer !== 0) {
+        engine.stopTimer(TraktorS2MK3.microphonePressedTimer);
+        TraktorS2MK3.microphonePressedTimer = 0;
+    }
+};
+
+TraktorS2MK3.parameterHandler = function (field) {
+    engine.setParameter(field.group, field.name, field.value / 4095);
+};
+
+TraktorS2MK3.jogTouchHandler = function (field) {
+    var deckNumber = TraktorS2MK3.controller.resolveDeck(field.group);
+    if (field.value > 0) {
+        engine.scratchEnable(deckNumber, 1024, 33.3333, 0.125, 0.125 / 8, true);
```

Comment:
```suggestion
        engine.scratchEnable(deckNumber, 1024, 33 + 1/3, 0.125, 0.125 / 8, true);
```
(That's what other mappings are doing)